File: Project Phoenix/Phoenix.py
import discord
import json
import os
import sys
from discord.ext import commands
from checks import has_access_level, has_permission, has_role
from commands import prefix, commands_dict

# set our internal version
version = "Phoenix"

# Define the file path to your JSON configuration file
config_file_path = "config.json"

# Load the JSON data from the file if it exists, or create a blank config if it doesn't exist
if os.path.exists(config_file_path):
    with open(config_file_path, "r") as config_file:
        config_data = json.load(config_file)
else:
    print(f"No configuration file found at {config_file_path}. Creating a blank configuration file...")
    config_data = {"Bot": {"owner": None, "token": None}}
    with open(config_file_path, "w") as config_file:
        json.dump(config_data, config_file)



# Define the intents
intents = discord.Intents.default()
client = discord.Client(intents=intents)

# Define the bot and set the command prefix and intents
bot = commands.Bot(command_prefix=prefix, intents=intents)

@client.event
async def on_ready():
    print("Bot is ready.")

# Define the commands
for command_name, command_properties in commands_dict.items():
    command_function = command_properties.get("function")
    command_aliases = command_properties.get("alias", [])
    command_access_level = command_properties.get("access_level")
    command_permission_needed = command_properties.get("permission_needed")
    command_role_needed = command_properties.get("role_needed")

    @bot.command(name=command_name, aliases=command_aliases)
    @has_access_level(command_access_level)
    @has_permission(command_permission_needed)
    @has_role(command_role_needed)
    async def command_wrapper(ctx):
        await command_function(ctx)

# Run the bot
client.run(config_data["Bot"]["token"])

File: Project Phoenix/Phoenix.js
const fs = require('fs');
const Discord = require('discord.js');

const { Client, Intents } = require('discord.js');
const intents = new Intents();
intents.add(Intents.FLAGS.GUILDS, Intents.FLAGS.GUILD_MESSAGES);
const client = new Client({ intents: intents });

const prefix = '!'; // Set your desired prefix here
client.commands = new Discord.Collection();

// Load the config file
const config = JSON.parse(fs.readFileSync('config.json', 'utf8'));

// Read all the command files from the commands folder
const commandFiles = fs.readdirSync('./commands').filter(file => file.endsWith('.js'));

for (const file of commandFiles) {
  const command = require(`./commands/${file}`);
  client.commands.set(command.name, command);
}

client.on('ready', () => {
  console.log(`Logged in as ${client.user.tag}!`);
});

client.on('message', message => {
  // Ignore messages sent by the bot itself
  if (message.author.bot) return;

  // Check if the message starts with the prefix
  if (message.content.startsWith(prefix)) {
    const args = message.content.slice(prefix.length).trim().split(/ +/);
    const commandName = args.shift().toLowerCase();

    // Get the command from the collection
    const command = client.commands.get(commandName)
      || client.commands.find(cmd => cmd.aliases && cmd.aliases.includes(commandName));

    if (!command) return;

    // Check if the user has the required role and permission
    if (command.roleNeeded && !message.member.roles.cache.some(role => role.name === command.roleNeeded)) {
      return message.reply(`you need the ${command.roleNeeded} role to use this command`);
    }
    if (command.permissionNeeded && !message.member.hasPermission(command.permissionNeeded)) {
      return message.reply(`you need the ${command.permissionNeeded} permission to use this command`);
    }

    // Execute the command
    try {
      command.execute(message, args);
    } catch (error) {
      console.error(error);
      message.reply('there was an error trying to execute that command');
    }
  }
});

// Login with the bot token
client.login(config.Bot.token);
